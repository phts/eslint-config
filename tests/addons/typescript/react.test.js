'use strict'

const helpers = require('../../helpers')

describe('addons/typescript/react', () => {
  helpers.itGeneratesCorrectConfig({
    extends: ['index', 'addons/typescript', 'addons/typescript/react'],
    files: ['src/target.ts', 'src/target.tsx', 'src/dir/target.ts', 'src/dir/target.tsx'],
    for: 'src/**/*.{ts,tsx}',
  })
})
