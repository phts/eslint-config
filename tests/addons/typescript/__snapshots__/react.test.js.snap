// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addons/typescript/react generates correct config for src/**/*.{ts,tsx} files 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
  },
  "globals": Object {},
  "ignorePatterns": Array [
    "playground/",
  ],
  "noInlineConfig": undefined,
  "parser": "node_modules/@typescript-eslint/parser/dist/index.js",
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": true,
    },
    "project": "./tsconfig.json",
    "sourceType": "module",
    "tsconfigRootDir": ".",
  },
  "plugins": Array [
    "eslint-comments",
    "import",
    "@typescript-eslint",
    "react",
  ],
  "reportUnusedDisableDirectives": undefined,
  "rules": Object {
    "@typescript-eslint/adjacent-overload-signatures": Array [
      "error",
    ],
    "@typescript-eslint/array-type": Array [
      "error",
      Object {
        "default": "array-simple",
      },
    ],
    "@typescript-eslint/ban-ts-comment": Array [
      "error",
    ],
    "@typescript-eslint/ban-ts-ignore": Array [
      "off",
    ],
    "@typescript-eslint/ban-types": Array [
      "error",
    ],
    "@typescript-eslint/consistent-type-assertions": Array [
      "error",
      Object {
        "assertionStyle": "as",
        "objectLiteralTypeAssertions": "allow-as-parameter",
      },
    ],
    "@typescript-eslint/consistent-type-definitions": Array [
      "error",
      "interface",
    ],
    "@typescript-eslint/explicit-function-return-type": Array [
      "off",
    ],
    "@typescript-eslint/explicit-member-accessibility": Array [
      "off",
    ],
    "@typescript-eslint/explicit-module-boundary-types": Array [
      "warn",
    ],
    "@typescript-eslint/indent": Array [
      "error",
      2,
    ],
    "@typescript-eslint/member-delimiter-style": Array [
      "error",
      Object {
        "multiline": Object {
          "delimiter": "none",
        },
        "singleline": Object {
          "delimiter": "comma",
          "requireLast": false,
        },
      },
    ],
    "@typescript-eslint/member-ordering": Array [
      "error",
    ],
    "@typescript-eslint/no-array-constructor": Array [
      "error",
    ],
    "@typescript-eslint/no-empty-function": Array [
      "error",
    ],
    "@typescript-eslint/no-empty-interface": Array [
      "off",
    ],
    "@typescript-eslint/no-explicit-any": Array [
      "error",
    ],
    "@typescript-eslint/no-extra-non-null-assertion": Array [
      "error",
    ],
    "@typescript-eslint/no-extra-semi": Array [
      "error",
    ],
    "@typescript-eslint/no-extraneous-class": Array [
      "error",
    ],
    "@typescript-eslint/no-inferrable-types": Array [
      "error",
    ],
    "@typescript-eslint/no-misused-new": Array [
      "error",
    ],
    "@typescript-eslint/no-namespace": Array [
      "error",
    ],
    "@typescript-eslint/no-non-null-asserted-optional-chain": Array [
      "error",
    ],
    "@typescript-eslint/no-non-null-assertion": Array [
      "off",
    ],
    "@typescript-eslint/no-parameter-properties": Array [
      "off",
    ],
    "@typescript-eslint/no-this-alias": Array [
      "error",
      Object {
        "allowDestructuring": true,
      },
    ],
    "@typescript-eslint/no-unused-vars": Array [
      "error",
    ],
    "@typescript-eslint/no-var-requires": Array [
      "error",
    ],
    "@typescript-eslint/prefer-as-const": Array [
      "error",
    ],
    "@typescript-eslint/prefer-namespace-keyword": Array [
      "error",
    ],
    "@typescript-eslint/restrict-plus-operands": Array [
      "error",
    ],
    "@typescript-eslint/triple-slash-reference": Array [
      "error",
    ],
    "accessor-pairs": Array [
      "error",
    ],
    "array-bracket-spacing": Array [
      "error",
    ],
    "brace-style": Array [
      "error",
    ],
    "camelcase": Array [
      "error",
    ],
    "comma-dangle": Array [
      "error",
      "always-multiline",
    ],
    "comma-spacing": Array [
      "error",
    ],
    "comma-style": Array [
      "error",
    ],
    "constructor-super": Array [
      "off",
    ],
    "curly": Array [
      "error",
    ],
    "dot-location": Array [
      "error",
      "property",
    ],
    "eol-last": Array [
      "error",
    ],
    "eqeqeq": Array [
      "error",
    ],
    "eslint-comments/disable-enable-pair": Array [
      "error",
      Object {
        "allowWholeFile": true,
      },
    ],
    "eslint-comments/no-aggregating-enable": Array [
      "error",
    ],
    "eslint-comments/no-duplicate-disable": Array [
      "error",
    ],
    "eslint-comments/no-unlimited-disable": Array [
      "error",
    ],
    "eslint-comments/no-unused-disable": Array [
      "error",
    ],
    "eslint-comments/no-unused-enable": Array [
      "error",
    ],
    "for-direction": Array [
      "error",
    ],
    "func-call-spacing": Array [
      "error",
    ],
    "getter-return": Array [
      "off",
    ],
    "import/export": Array [
      "error",
    ],
    "import/first": Array [
      "error",
    ],
    "import/named": Array [
      "off",
    ],
    "import/newline-after-import": Array [
      "error",
    ],
    "import/no-absolute-path": Array [
      "error",
    ],
    "import/no-cycle": Array [
      "error",
    ],
    "import/no-named-as-default": Array [
      "error",
    ],
    "import/no-named-as-default-member": Array [
      "error",
    ],
    "import/no-self-import": Array [
      "error",
    ],
    "import/no-useless-path-segments": Array [
      "error",
    ],
    "import/order": Array [
      "error",
    ],
    "indent": Array [
      "off",
      2,
    ],
    "jsx-quotes": Array [
      "error",
      "prefer-double",
    ],
    "key-spacing": Array [
      "error",
      Object {
        "afterColon": true,
        "beforeColon": false,
        "mode": "minimum",
      },
    ],
    "keyword-spacing": Array [
      "error",
    ],
    "max-len": Array [
      "error",
      Object {
        "code": 100,
      },
    ],
    "new-cap": Array [
      "error",
    ],
    "new-parens": Array [
      "error",
    ],
    "no-array-constructor": Array [
      "off",
    ],
    "no-async-promise-executor": Array [
      "error",
    ],
    "no-caller": Array [
      "error",
    ],
    "no-case-declarations": Array [
      "error",
    ],
    "no-class-assign": Array [
      "error",
    ],
    "no-compare-neg-zero": Array [
      "error",
    ],
    "no-cond-assign": Array [
      "error",
    ],
    "no-console": Array [
      "error",
      Object {
        "allow": Array [
          "info",
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": Array [
      "off",
    ],
    "no-constant-condition": Array [
      "error",
    ],
    "no-control-regex": Array [
      "error",
    ],
    "no-debugger": Array [
      "error",
    ],
    "no-delete-var": Array [
      "error",
    ],
    "no-dupe-args": Array [
      "off",
    ],
    "no-dupe-class-members": Array [
      "off",
    ],
    "no-dupe-else-if": Array [
      "error",
    ],
    "no-dupe-keys": Array [
      "off",
    ],
    "no-duplicate-case": Array [
      "error",
    ],
    "no-duplicate-imports": Array [
      "error",
    ],
    "no-empty": Array [
      "error",
    ],
    "no-empty-character-class": Array [
      "error",
    ],
    "no-empty-function": Array [
      "off",
    ],
    "no-empty-pattern": Array [
      "error",
    ],
    "no-eval": Array [
      "error",
    ],
    "no-ex-assign": Array [
      "error",
    ],
    "no-extra-bind": Array [
      "error",
    ],
    "no-extra-boolean-cast": Array [
      "error",
    ],
    "no-extra-parens": Array [
      "off",
      "all",
      Object {
        "ignoreJSX": "all",
      },
    ],
    "no-extra-semi": Array [
      "off",
    ],
    "no-fallthrough": Array [
      "error",
    ],
    "no-floating-decimal": Array [
      "error",
    ],
    "no-func-assign": Array [
      "off",
    ],
    "no-global-assign": Array [
      "error",
    ],
    "no-implied-eval": Array [
      "error",
    ],
    "no-import-assign": Array [
      "off",
    ],
    "no-inner-declarations": Array [
      "error",
    ],
    "no-invalid-regexp": Array [
      "error",
    ],
    "no-irregular-whitespace": Array [
      "error",
    ],
    "no-label-var": Array [
      "error",
    ],
    "no-labels": Array [
      "error",
    ],
    "no-lone-blocks": Array [
      "error",
    ],
    "no-misleading-character-class": Array [
      "error",
    ],
    "no-mixed-spaces-and-tabs": Array [
      "error",
    ],
    "no-multiple-empty-lines": Array [
      "error",
    ],
    "no-new-object": Array [
      "error",
    ],
    "no-new-require": Array [
      "error",
    ],
    "no-new-symbol": Array [
      "off",
    ],
    "no-obj-calls": Array [
      "off",
    ],
    "no-octal": Array [
      "error",
    ],
    "no-path-concat": Array [
      "error",
    ],
    "no-proto": Array [
      "error",
    ],
    "no-prototype-builtins": Array [
      "error",
    ],
    "no-redeclare": Array [
      "off",
    ],
    "no-regex-spaces": Array [
      "error",
    ],
    "no-return-assign": Array [
      "error",
    ],
    "no-return-await": Array [
      "error",
    ],
    "no-self-assign": Array [
      "error",
    ],
    "no-self-compare": Array [
      "error",
    ],
    "no-sequences": Array [
      "error",
    ],
    "no-setter-return": Array [
      "off",
    ],
    "no-shadow-restricted-names": Array [
      "error",
    ],
    "no-sparse-arrays": Array [
      "error",
    ],
    "no-tabs": Array [
      "error",
    ],
    "no-this-before-super": Array [
      "off",
    ],
    "no-throw-literal": Array [
      "error",
    ],
    "no-trailing-spaces": Array [
      "error",
    ],
    "no-undef": Array [
      "off",
    ],
    "no-undef-init": Array [
      "error",
    ],
    "no-unexpected-multiline": Array [
      "error",
    ],
    "no-unmodified-loop-condition": Array [
      "error",
    ],
    "no-unneeded-ternary": Array [
      "error",
    ],
    "no-unreachable": Array [
      "off",
    ],
    "no-unsafe-finally": Array [
      "error",
    ],
    "no-unsafe-negation": Array [
      "off",
    ],
    "no-unused-labels": Array [
      "error",
    ],
    "no-unused-vars": Array [
      "off",
    ],
    "no-use-before-define": Array [
      "error",
    ],
    "no-useless-call": Array [
      "error",
    ],
    "no-useless-catch": Array [
      "error",
    ],
    "no-useless-computed-key": Array [
      "error",
    ],
    "no-useless-constructor": Array [
      "error",
    ],
    "no-useless-escape": Array [
      "error",
    ],
    "no-useless-rename": Array [
      "error",
    ],
    "no-var": Array [
      "error",
    ],
    "no-whitespace-before-property": Array [
      "error",
    ],
    "no-with": Array [
      "error",
    ],
    "object-curly-spacing": Array [
      "error",
      "never",
    ],
    "object-property-newline": Array [
      "error",
      Object {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false,
      },
    ],
    "one-var": Array [
      "error",
      "never",
    ],
    "one-var-declaration-per-line": Array [
      "error",
    ],
    "padded-blocks": Array [
      "error",
      "never",
    ],
    "prefer-const": Array [
      "error",
    ],
    "prefer-rest-params": Array [
      "error",
    ],
    "prefer-spread": Array [
      "error",
    ],
    "quote-props": Array [
      "error",
      "as-needed",
    ],
    "quotes": Array [
      "error",
      "single",
      Object {
        "allowTemplateLiterals": false,
        "avoidEscape": true,
      },
    ],
    "react/display-name": Array [
      "off",
    ],
    "react/jsx-key": Array [
      2,
    ],
    "react/jsx-no-comment-textnodes": Array [
      2,
    ],
    "react/jsx-no-duplicate-props": Array [
      2,
    ],
    "react/jsx-no-target-blank": Array [
      2,
    ],
    "react/jsx-no-undef": Array [
      2,
    ],
    "react/jsx-uses-react": Array [
      2,
    ],
    "react/jsx-uses-vars": Array [
      2,
    ],
    "react/no-children-prop": Array [
      2,
    ],
    "react/no-danger-with-children": Array [
      2,
    ],
    "react/no-deprecated": Array [
      2,
    ],
    "react/no-direct-mutation-state": Array [
      2,
    ],
    "react/no-find-dom-node": Array [
      2,
    ],
    "react/no-is-mounted": Array [
      2,
    ],
    "react/no-render-return-value": Array [
      2,
    ],
    "react/no-string-refs": Array [
      2,
    ],
    "react/no-unescaped-entities": Array [
      2,
    ],
    "react/no-unknown-property": Array [
      2,
    ],
    "react/no-unsafe": Array [
      0,
    ],
    "react/prop-types": Array [
      "off",
    ],
    "react/react-in-jsx-scope": Array [
      2,
    ],
    "react/require-render-return": Array [
      2,
    ],
    "require-yield": Array [
      "error",
    ],
    "rest-spread-spacing": Array [
      "error",
    ],
    "semi": Array [
      "error",
      "never",
    ],
    "semi-spacing": Array [
      "error",
    ],
    "space-before-blocks": Array [
      "error",
    ],
    "space-before-function-paren": Array [
      "error",
      Object {
        "anonymous": "always",
        "asyncArrow": "always",
        "named": "never",
      },
    ],
    "space-in-parens": Array [
      "error",
    ],
    "space-infix-ops": Array [
      "error",
    ],
    "space-unary-ops": Array [
      "error",
    ],
    "spaced-comment": Array [
      "error",
    ],
    "strict": Array [
      "error",
    ],
    "template-curly-spacing": Array [
      "error",
    ],
    "use-isnan": Array [
      "error",
    ],
    "valid-typeof": Array [
      "off",
    ],
    "wrap-iife": Array [
      "error",
    ],
    "yield-star-spacing": Array [
      "error",
      "before",
    ],
    "yoda": Array [
      "error",
    ],
  },
  "settings": Object {
    "react": Object {
      "version": "detect",
    },
  },
}
`;
